### PR TITLE
TRUNK-4782 : Retrospective edit of Orders

### PR DESCRIPTION
**Pull request by aj-jaswanth**
Hi @djazayeri, 
This is an initial commit. Want to have this as a starting point for the discussion. More tests have to be written though. :)

### CODE REVIEW

**Comment by aj-jaswanth** [comment]
@djazayeri Is this approach okay?

**Comment by angshu** [comment]
@aj-jaswanth shouldn't this fix be applicable to 1.12 branch as well?

**Comment by aj-jaswanth** [comment]
@angshu Yes. I am improving the code. Will raise a pull request on 1.12.x.

File: api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -88,6 +88,18 @@ public void setOrderDAO(OrderDAO dao) {
...
+
+	/**
+	 * @see org.openmrs.api.OrderService#saveOrder(org.openmrs.Order, org.openmrs.api.OrderContext)
+	 */
+	@Override
+	public synchronized Order saveRetrospectiveOrder(Order order, OrderContext orderContext) {
+		return saveOrder0(order, orderContext, true);
+	}
+
+	private Order saveOrder0(Order order, OrderContext orderContext, boolean isRetrospective) {
```
Comment (djazayeri) [inline api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java]: this method name is not good enough

**Comment by aj-jaswanth** [comment]
@djazayeri https://github.com/openmrs/openmrs-core/pull/1635 Raised another PR.

**Pull request closed without merging**